### ISSUE TITLE
updates

### ISSUE THREAD

**Issue by jandegr** [zoff99/zanavi#31]
Hi @zoff99 

I made some updates https://github.com/jandegr/zanavi/issues/2

**Comment by zoff99** [comment]
@jandegr yeah i saw it. does it work fully? lets talk more on IRC if you have some time

**Comment by jandegr** [comment]
@zoff99 I will hang around in #zanavi irc from time to time

**Comment by jandegr** [comment]
@zoff99 
Any chance (some of) the maps could get an update since the mapserver(s) are still alive anyway.
I can build maps for my own so I don't really need it for myself but if the maps are not updated then I don't need to preserve the functionality of the mapdownloader. In the ohter case I would keep the mapdowloader working and just maybe my updated version can be used again if somebody is interested.

THX

**Comment by jandegr** [comment]
I rebuilt a bunch of maps on CI and tweaked the mapdownloader a bit so it can fetch those directly from there. Unfortunatly the larger ones (Germany, France) can only be built locally since the runtime exceeds 5 hours on CI, so those are not available for download from some server.

Any chance the rebuilt maps can find their way onto the Zanavi mapserver ?
If so, more of them can be rebuilt.

thx.

![Screenshot_1575626633](https://user-images.githubusercontent.com/9937104/70315058-168fe700-1819-11ea-97f9-e39a0f826365.png)

**Comment by zoff99** [comment]
@jandegr what is the exact command line you use to build the maps?

**Comment by jandegr** [comment]
Hi,
Here is the attempt I made for Germany.
https://circleci.com/gh/jandegr/zanavi/258#config/containers/0
Whenever I increase slice size or jobs I get an out of memory after a while. A box has 4GB and 2 vcores on CI.
Locally I could build France with some patience, so there is no lack of maps for me but it is more practical if maps are available for download from somewhere and for everybody, so CI was a convenient solution.

For the rest I just added 3 extra attrributes and had to increase this https://github.com/jandegr/zanavi/blob/7153dd3f6301918fe6faf6bc79a0d3aac16e52a8/navit/maptool/osm.c#L5956 

THX.

**Comment by zoff99** [comment]
can you post the exact command to build your maps here please?

**Comment by jandegr** [comment]
I tried some variations of slice size and jobs, but here is the one I last tried for Germany
`bzcat map1.osm.bz2 | ./maptool -o runtimes.sqlite -6 -k -n -U -S 425000000 map.bin`

For a local builds I used 2GB slice size but then ther is no time limit either.

EDIT: I am testing one without the -o runtimes.sqlite switch now. I copied that over from https://github.com/zoff99/zanavi_maptool/blob/master/circle.yml but it seems to run faster without.

**Comment by zoff99** [comment]
```
INFILE="example.osm.bz2"
OUT_BINFILE="example.bin"
MEG_SIZE=1880 # im MByte
BYTE_SIZE="$MEG_SIZE""000000"
FIXED_COUNTRY_ID=""

mkdir -p db
mkdir -p db2
mkdir -p db3
mkdir -p manual

bzip2 -dc $INFILE | \
maptool -n -f -x -U -j 1 -o maptool_runtimes.sqlite \
-S ${BYTE_SIZE} ${FIXED_COUNTRY_ID} ${OUT_BINFILE}

```

**Comment by zoff99** [comment]
there is much more to it, especially for large maps you need different options.

**Comment by zoff99** [comment]
did you compile the latested version of maptool?
